docs/contribution-guide.md

```diff
@@ -1,19 +1,104 @@
+[apidoc]: https://kiltprotocol.github.io/sdk-js/api/index.html
+[jsdoc]: https://www.npmjs.com/package/eslint-plugin-jsdoc
+[typedoc]: https://github.com/TypeStrong/typedoc
+
 # Contribution Guide
 
-## NB
+## Documentation
+
+We want the KILT SDK to be easy to use for all fellow developers. We ❤️ well-documented code.
+
+KILT's API doc is available [here][apidoc].
+
+In the KILT SDK, modules and public methods are documented inline, in the code. These inline "doc blocks" need to follow some linting rules. Make sure ESLint is activated in your code editor, so the errors and warnings are highlighted.
+
+We also check these rules on commit, so that you're protected: you can't push a linting error to the repo.
+
+You don't need to - but if you're interested, you can see all the linting rules for the inline doc blocks in `.eslintrc.json`, and check what they mean on [jsdoc][jsdoc].
+
+We're using [typedoc][typedoc] to generate the API doc from the inline doc blocks.
+
+### Documenting modules
+
+Direct child folders of `src` such as `attestation` should be marked as `@modules`, so that they're listed in the [online API Doc][apidoc] main menu.
+
+Some of these modules are purely technical utilities, such as `crypto`. Others map to KILT concepts, such as `attestation`.
+
+Both types of modules shall be documented with a short text in one of their child files (usually `X.ts` if the folder is called `x`):
+
+* For technical modules, you're free to write this text as you like;
+* For modules that map to a KILT concept, we recommend that you structure your text as follows:
+  * Paragraph 1: Explanation of the overall concept, and the role it fulfills in the KILT Protocol;
+  * Paragraph 2: Technical details of interest.
+
+For reference, look for example at the doc block in `Attestation.ts`.
+
+### Documenting public methods
+
+Since they're available to SDK users, public methods must be documented.
+
+On top of the the linting rules mentioned above (must document all parameters, must have a description...), we recommend that you observe the following guidelines to make the documentation as helpful as possible:
+
+* Method description:
+  * Make it concise and clear;
+  * Start with a capitalized verb in the 3rd person;
+  * If the method is async and/or static, add [ASYNC] and/or [STATIC] right before the description content.
+* `@param` and `@returns` fields: don't add types, since this is automatically added into the API doc;
+* Make sure you explain opaque abbreviations or jargon (example: TxStatus = transaction status);
+* When referring to SDK Classes and methods, make sure you link them in, using `[[]]`;
+* `@example`:
+  * Create it as valid code 😎;
+  * Illustrate only this method's functionality, but also provide enough context for fellow developers to try it out:
+    * Imports;
+    * Preparation step;
+    * Actual function call;
+    * Expected output;
+    * Suggestion for the next step;
+    * ...
+  * Don't hesitate to include comments.
```

Comment:
Let's also say, that this doesn't have to be overdone. Just a bit of needed context to understand the method call is enough